### ISSUE TITLE
I would like to know what the purpose of this code

### ISSUE THREAD

**Issue by oner0916** [RUCAIBox/UniSRec#12]
I would like to know what the purpose of this code:
path=`pwd`
for ds in Food Home CDs Kindle Movies
do
  ln -s ${path}/../downstream/${ds}/${ds}.feat1CLS ../pretrain/FHCKM/
  ln -s ${path}/../downstream/${ds}/${ds}.feat2CLS ../pretrain/FHCKM/
done

**Comment by oner0916** [comment]
Is there something wrong? Or the "downstream" should be the "pretrain"?

**Comment by hyp1231** [comment]
Hi, thanks for your attention!

The purpose is to create some soft links in the `pretrain` dir. Take the "Food" dataset for an example, we originally created `Food.feat1CLS` in the `downstream/Food` dir, containing the BERT representations for item text in the "Food" domain. As we also need these BERT representations in the pre-training stage, for convenience, we just add some soft links in `pretrain/FHCKM` to access these BERT representations that physically exist in there original `downstream/${ds}` dirs.

**Comment by oner0916** [comment]
Dear Dr.Hou:
Thank you for you reply. Regarding this issue, I have another question. Take the "Food" dataset for an example, "Food" dataset is a pre-training dataset. So, we execute the following command: "python process_amazon.py --dataset ${ds} --output_path ../pretrain/ --word_drop_ratio 0.15" for data preprocessing.&nbsp;The result of this command is to create a directory in "pretrain/Food/" to store the processed data, rather than in the downstream/Food directory.&nbsp; So regarding what you mentioned about "Food.feat1CLS in the downstream/Food directory," I am quite puzzled.


I would greatly appreciate your response. Thank you in advance for your assistance.


Best Regards,,
Yihong Lu



ONER
***@***.***

华南师范大学软件学院


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "RUCAIBox/UniSRec"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月21日(星期四) 凌晨2:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [RUCAIBox/UniSRec] I would like to know what the purpose of this code (Issue #12)





 
Hi, thanks for your attention!
 
The purpose is to create some soft links in the pretrain dir. Take the "Food" dataset for an example, we originally created Food.feat1CLS in the downstream/Food dir, containing the BERT representations for item text in the "Food" domain. As we also need these BERT representations in the pre-training stage, for convenience, we just add some soft links in pretrain/FHCKM to access these BERT representations that physically exist in there original downstream/${ds} dirs.

**Comment by hyp1231** [comment]
Hi Yihong,

Thanks!! I got the point. You are right, and there is indeed a bug here. I'll update the script later and thanks a lot for pointing this!

By the way, as I've not received one PhD degree, I cannot use the "Dr." prefix. Just call me Yupeng, please. Thanks!

Best,
Yupeng

**Comment by oner0916** [comment]
Dear Yupeng,
I'm glad to receive your response. I have two questions about the code that I'd like to ask you. 1）I'd like to know what dataset filtering does (I didn't quite understand your source code). 2）I'd like to know what `self.field2id_token['item_id']` represents because it seems that there isn't a specific assignment process in the code. 

&nbsp;I would greatly appreciate your help and guidance in this matter.  I look forward to hearing from you soon and am sincerely grateful for your time and consideration.


Sincerely,
Yihong Lu 




ONER
***@***.***

华南师范大学软件学院


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "RUCAIBox/UniSRec"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月25日(星期一) 上午10:32
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [RUCAIBox/UniSRec] I would like to know what the purpose of this code (Issue #12)





 
Hi Yihong,
 
Thanks!! I got the point. You are right, and there is indeed a bug here. I'll update the script later and thanks a lot for pointing this!
 
By the way, as I've not received one PhD degree, I cannot use the "Dr." prefix. Just call me Yupeng, please. Thanks!
 
Best,
 Yupeng

**Comment by hyp1231** [comment]
1) dataset filtering.
We filter out items without metadata, then filter out users/items under 5 interactions (5-core filtering).

2) `self.field2id_token['item_id']`
Yes the attribute is defined in the original RecBole library. One can refer to RecBole [API Doc](https://recbole.io/docs/recbole/recbole.data.dataset.dataset.html#recbole.data.dataset.dataset.Dataset.field2id_token):
> Dict mapping feature name (str) to a `np.ndarray`, which stores the original token of this feature. For example, if `item_id` is token-like feature, `token_a` is remapped to 1, `token_b` is remapped to 2. Then field2id_token['item_id'] = ['[PAD]', 'token_a', 'token_b']. (Note that 0 is always PADDING for token-like features.) Type: dict
